File: Java/BinarySearch.java
public class BinarySearch {

    static int binarySearch(int arr[], int low, int high, int x) {
        while (low <= high) {
            int mid = low + (high - low) / 2;

            if (arr[mid] == x)
                return mid;
            else if (arr[mid] < x)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return -1;
    }

    public static void main(String[] args) {
        int arr[] = {2, 3, 4, 10, 40};
        int x = 10;
        int result = binarySearch(arr, 0, arr.length - 1, x);

        if (result != -1)
            System.out.println("Element is present at index " + result);
        else
            System.out.println("Element is not present in array");
    }
}

File: BinarySearch.py
# Iterative
def binary_search_iter(arr, x):
    low, high = 0, len(arr)-1
    while low <= high:
        mid = (low + high)//2
        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            low = mid + 1
        else:
            high = mid - 1
    return -1


# Recursive
def binary_search_rec(arr, low, high, x):
    if low <= high:
        mid = (low + high)//2
        if arr[mid] == x:
            return mid
        elif arr[mid] < x:
            return binary_search_rec(arr, mid+1, high, x)
        else:
            return binary_search_rec(arr, low, mid-1, x)
    return -1


arr = sorted(list(map(int, input("Enter sorted array: ").split())))
x = int(input("Enter element to search: "))

print("Iterative Index:", binary_search_iter(arr, x))
print("Recursive Index:", binary_search_rec(arr, 0, len(arr)-1, x))